Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -877,9 +877,10 @@ public final class PackageBuilder {
                 }
 
                 // Ensure that the search path is contained within the package.
-                let subpath = try RelativePath(validating: value)
-                guard targetRoot.appending(subpath).isDescendantOfOrEqual(to: packagePath) else {
-                    throw ModuleError.invalidHeaderSearchPath(subpath.pathString)
+                _ = try RelativePath(validating: value)
```

Comment:
same comment as https://github.com/apple/swift-package-manager/pull/5484/files#r870728183

this change is too subtle and likely to be reverted by future maintainers as the intuitive thing to do is to note ignore the relative path returned by the constructor. we really do need to address the underlying issue first.